cpp/src/io/orc/writer_impl.cu

```diff
@@ -2702,7 +2703,8 @@ void writer::impl::close()
   ps.footerLength         = pbw.size();
   ps.compression          = _compression_kind;
   ps.compressionBlockSize = _compression_blocksize;
-  ps.version              = {0, 12};
+  ps.version              = {0, 12};  // Hive 0.12
+  ps.writerVersion        = 7;        // ORC-517 fixed
```

Comment:
should this `7` be defined above like `MAGIC` with a comment about it? Do we change the writer version every major release or major changes? Is this something we never change?

Reply:
My understanding is that we would increase this version when we implement newer features and fix notable bugs, see https://github.com/apache/orc/blob/main/proto/orc_proto.proto#L412-L422

I'm not sure if this should be a constant. I'll make the change if you think that's better.

Reply:
If it is a constant it is easier to add a multi-line comment explaining that major changes should increment it. I'd also be ok with a comment about the version. At the extreme, I would have a changelist in the comments
```
// version 1: Initial implementation
// version 2: added nested types
// version 3: fixed bug xyz
```

Reply:
Included the list of versions, let me know what you think